test/powershell/engine/PropertyAccessor.Tests.ps1

```diff
@@ -0,0 +1,96 @@
+#
+# Functional tests to verify basic conditions for IO to the PowerShellProperties.json files
+# The properties files are supported on non-Windows OSes, but the tests are specific to 
+# Windows so that file IO can be verified using supported cmdlets.
+# 
+
+try {
+
+    #skip all tests on non-windows platform
+    $originalDefaultParameterValues = $PSDefaultParameterValues.Clone()
+    $IsNotSkipped = ($IsWindows -eq $true);
+    $PSDefaultParameterValues["it:skip"] = !$IsNotSkipped
+    $ShouldSkipTest = !$IsNotSkipped
+
+    Describe "User-Specific PowerShellProperties.json Modifications" -Tags "CI" {
+
+        BeforeAll {
+            if ($IsNotSkipped) {
+                # Discover the user-specific PowerShellProperties.json file
+                $splitModulePath = $env:PSMODULEPATH -split ";"
+                $userModuleDir = $splitModulePath[0].TrimEnd("\")
+                $userSettingsDir = $userModuleDir -replace "\\Modules", ""
+                $userPropertiesFile = Join-Path $userSettingsDir "PowerShellProperties.json"
+
+                # Save the file for restoration after the tests are complete
+                $backupPropertiesFile = ""
+                if (Test-Path $userPropertiesFile) {
+                    $backupPropertiesFile = Join-Path $userSettingsDir "ORIGINAL_PowerShellProperties.json"
+                    Copy-Item -Path $userPropertiesFile -Destination $backupPropertiesFile -Force -ErrorAction Continue
+                }
+                else {
+                    # create the directory and file if they do not already exist
+                    $null = New-Item -Type Directory  -Path $userSettingsDir -Force -ErrorAction SilentlyContinue
```

Comment:
Do we want to create the directory `C:\Users\dongbow\Documents` when it already exists?
Probably it should be
``` powershell
elseif (-not (Test-Path $userSettingsDir))
{
    $null = New-Item -Type Directory -Path $userSettingsDir -Force -ErrorAction SilentlyContinue
}
```